src/Language/Mulang/Inspector/Generic.hs

```diff
@@ -69,12 +69,21 @@ assignsMatching matcher predicate = containsExpression f
   where f (Unification name value) = predicate name && matcher [value]
         f _                        = False
 
+assignsFieldMatching :: Matcher -> BoundInspection
+assignsFieldMatching matcher predicate = containsExpression f
+  where f (FieldAssignment _ name value) = predicate name && matcher [value]
+        f _                              = False
+
 -- | Inspection that tells whether an expression uses the the given target identifier
 -- in its definition
 uses :: BoundInspection
 uses p = containsExpression f
   where f = any p . referencedIdentifiers
 
+usesField :: BoundInspection
+usesField p = containsExpression f
```

Comment:
I have two concerns regarding this:
- Should the definition be the exact same as `uses`'? What if I just use a variable with the same name as a field, and never access the field itself?
- We don't seem to be exporting this in the module (`assignsFieldMatching` either, for that matter)

Reply:
All this function look like I just copy pasted the `uses` inspection. It should call `assignsFieldMatching` instead.

Reply:
In order to continue blocking this issue, I am moving the new inspections to another branch.